File: catmull_rom_t.m
function [q] = catmull_rom_t(control_x, control_y, T)
  q = @(t) 1/2*[t^3 t^2 t 1]*[-1/T (4*T-1)/T (-4*T +1)/T 1/T; 2/T (-6*T+1)/T (6*T-2)/T -1/T; -1/T 0 1/T 0; 0 2 0 0]*[control_x' control_y'];
end

File: src/modules/catmull_rom_t.py
import numpy as np

class CatmullRomT:
    def __init__(self, control_x, control_y, T):

        self.control_x = control_x
        self.control_y = control_y
        self.T = T
        self.q = self.catmull_rom_t()



    def catmull_rom_t(self):
        def q(t):
            return 1/2 * np.array([t**3, t**2, t, 1]) @ np.array([[-1/self.T, (4*self.T-1)/self.T, (-4*self.T +1)/self.T, 1/self.T], 
                                                                  [2/self.T, (-6*self.T+1)/self.T, (6*self.T-2)/self.T, -1/self.T], 
                                                                  [-1/self.T, 0, 1/self.T, 0], 
                                                                  [0, 2, 0, 0]]) @ np.array([self.control_x, self.control_y]).T
        return q
    
    

